### DISCUSSION TITLE
Deribit Sell/Buy functions

### DISCUSSION THREAD

**Discussion by onurakinsir** [cafawo/FinancialDataAnalytics#19] [Q&A] [answered]
My functions, sell and buy, were working perfectly – however, now I am getting errors when I execute the function.

```
      9 response = requests.get(url, params=params, auth=(ClientID, ClientSecret))
     10 response_json = response.json()
---> 11 response_json['result']
     12 print(f"{selling_amount} {instrument_name} were sold for ${P_0}.")

KeyError: 'result'
```

**Comment by cafawo** [comment]
Two points on your issue:

1. You might have to slow down on the frequency, e.g. using ``from time import sleep``, see: https://www.deribit.com/kb/deribit-rate-limits

2. You should check if your request was successful before trying to access ``response_json['result']``, because you can get information on why your request failed from ``response_json``.

**Reply by onurakinsir** [reply]
Thanks for your answer! 

1. I try to sleep for 1-2 seconds for a request – it did not work either.
2. `Failed to retrieve data: 400` 

I also try to change my ClientID and ClientSecret by creating a new API key. But unfortunately, it did not work either.

**Comment by onurakinsir** [comment, marked as answer]
Finally, I solved the issue. The problem is when I call the function. I have been setting the amount parameter as one (1). It perfectly works if I work on _ETH_, however, when it comes to _BTC_ it does not work at all. I had to change the requested order size (amount) from 1 to 10 for _BTC_ (at least).
